Packs/Arcanna/Integrations/ArcannaAI/ArcannaAI.py

```diff
@@ -13,342 +16,411 @@
+		raw_response = requests.get(url=self.base_url + url_suffix, headers=self.get_headers(), verify=self.verify)
+		if raw_response.status_code not in [200, 422, 500]:
+			 raise Exception(f"Error HttpCode={raw_response.status_code} text={raw_response.text} Job ID :#{self.base_url}# #{url_suffix}# #{self.get_headers()}# #{job_id}# #{body}#")
+		return raw_response.json() 
+		
+	def trigger_training(self, job_id, username):
+		url_suffix = f"/api/v1/jobs/{job_id}/train?username={username}"
+		body = {
+			 "username": username
+		}
+		raw_response = requests.post(url=self.base_url + url_suffix, headers=self.get_headers(), verify=self.verify, json=body)
+		if raw_response.status_code not in [200, 422, 500]:
+			 raise Exception(f"Error HttpCode={raw_response.status_code} text={raw_response.text} Job ID :#{self.base_url}# #{url_suffix}# #{self.get_headers()}# #{job_id}# #{body}#")
+		return raw_response.json() 
+	
+	def get_decision_set(self, job_id):
+		url_suffix = f"/api/v1/jobs/{job_id}/labels"
+		raw_response = requests.get(url=self.base_url + url_suffix, headers=self.get_headers(), verify=self.verify)
+		if raw_response.status_code not in [200, 422, 500]:
+			 raise Exception(f"Error HttpCode={raw_response.status_code} text={raw_response.text} Job ID :#{self.base_url}# #{url_suffix}# #{self.get_headers()}# #{job_id}# #{body}#")
+		return raw_response.json() 
+
+	def send_raw_event(self, job_id, severity, title, raw_body,id_value=None):
+		url_suffix = 'api/v1/events/'
+		if id_value is not None and id_value != "":
+			url_suffix = f"{url_suffix}{id_value}"
+		raw = json.loads(raw_body)
+		body = self.map_to_arcanna_raw_event(job_id, raw, severity, title)
+		raw_response = None
+		raw_response = requests.post(url=self.base_url + url_suffix, headers=self.get_headers(), verify=self.verify,
+									 json=body)
+		if raw_response.status_code not in [201,422,500]:
+			raise Exception(f"Error HttpCode={raw_response.status_code} text={raw_response.text} Job ID :#{self.base_url}# #{url_suffix}# #{self.get_headers()}# #{job_id}# #{body}#")
+		
+		return raw_response.json()
+
+	def map_to_arcanna_raw_event(self, job_id, raw, severity, title):
+		body = {
+			"job_id": int(job_id),
+			"title": title,
+			"raw_body": raw
+		}
+		if severity is not None:
+			severity = int(float(severity))
+			body["severity"] = severity
+		return body
+
+	def get_event_status(self, job_id, event_id, retries=10, seconds_per_retry=3):
+		retry_count = 0
+		while(retry_count < retries):
+			url_suffix = f"api/v1/events/{job_id}/{event_id}"
+			full_url = f"{self.base_url}{url_suffix}"
+			raw_response = requests.get(url=full_url, headers=self.get_headers(), verify=self.verify)
+			if raw_response.status_code not in [200,422]:
+				raise Exception(f"Error HttpCode={raw_response.status_code} text={raw_response.text} Job ID :#{self.base_url}# #{url_suffix}#")
+			if raw_response.json()['status'] == 'pending_inference':
+				retry_count += 1
+				time.sleep(seconds_per_retry)
```

Comment:
Do not use sleep in your code.
Please see the following approach - https://xsoar.pan.dev/docs/tutorials/tut-design#command-design

Reply:
removed sleep and retry mechanism as there's no support for retrying a step that has valid response ( _http_request has the retry mechanism but it's limited to status codes , not being able to use the remote response as a retry mechanism condition, at least from what I read in github for that class )